app/core/store/modules/seasonalLeague.js

```diff
@@ -0,0 +1,215 @@
+            rank -= 1
+            aboveSession.rank = rank
+          }
+          playersAbove.reverse()
+          rank = parseInt(myRank, 10)
+          for (const belowSession of playersBelow) {
+            rank += 1
+            belowSession.rank = rank
+          }
+
+          teamSession.rank = parseInt(myRank, 10)
+          commit('setMySession', teamSession)
+          commit('setGlobalAbove', playersAbove)
+          commit('setGlobalBelow', playersBelow)
+        }
+      }
+      await Promise.all(awaitPromises)
+      commit('setLoading', false)
+    },
+
+    async loadClanRequiredData ({ commit }, { leagueId }) {
+      const leagueRankingInfo = {
+        top: [],
+        playersAbove: [],
+        playersBelow: []
+      }
+
+      const topLeagueRankingPromise = getLeaderboard(currentSeasonalLevelOriginal, {
+        order: -1,
+        scoreOffset: 1000000,
+        limit: 20,
+        team: 'humans',
+        'leagues.leagueID': leagueId,
+        '_': Math.floor(Math.random() * 100000000)
+      }).then(ranking => {
+        leagueRankingInfo.top = ranking
+      })
+
+      const sessionsData = await fetchMySessions(currentSeasonalLevelOriginal)
+
+      if (Array.isArray(sessionsData) && sessionsData.length > 0) {
+        const teamSession = sessionsData.find((session) => session.team === 'humans')
+        if (!teamSession) {
+          commit('setLoading', false)
+          return
+        }
+        const score = (((teamSession.leagues || []).find(({ leagueID }) => leagueID === leagueId) || {}).stats || {}).totalScore;
+
+        if (score !== undefined) {
+          const [ playersAbove, playersBelow, myRank ] = await Promise.all([
+            getLeaderboard(currentSeasonalLevelOriginal, { order: 1, scoreOffset: score, limit: 4, 'leagues.leagueID': leagueId }),
+            getLeaderboard(currentSeasonalLevelOriginal, { order: -1, scoreOffset: score, limit: 4, 'leagues.leagueID': leagueId }),
+            getMyRank(currentSeasonalLevelOriginal, teamSession._id, {
+              scoreOffset: score,
+              team: 'humans',
+              'leagues.leagueID': leagueId
+            })
+          ])
+
+          let rank = parseInt(myRank, 10)
+          for (const aboveSession of playersAbove) {
+            rank -= 1
+            aboveSession.rank = rank
+          }
+          playersAbove.reverse()
+          rank = parseInt(myRank, 10)
+          for (const belowSession of playersBelow) {
+            rank += 1
+            belowSession.rank = rank
+          }
+
+          teamSession.rank = parseInt(myRank, 10)
+          leagueRankingInfo.playersAbove = playersAbove
+          leagueRankingInfo.playersBelow = playersBelow
+
+          commit('setMySession', teamSession)
+        }
+      }
+
+      await topLeagueRankingPromise
+
+      commit('setLeagueRanking', { leagueId: leagueId, ranking: leagueRankingInfo })
+    },
+
+    async fetchGlobalLeaderboard ({ commit }) {
```

Comment:
It seems like we have some duplicated logic between the global case and the clan case. Do we need to have the code paths be so separate? Everything is and likely will be designed to remain parallel, and the server response is going to just depend on the presence of a `leagues.leagueID` parameter. We also don't show global and league leaderboards together at the same time. Seems like we should have this logic be one code path that just differs by whether we're passing in a league.

Reply:
I agree we could merge them together in the future. I implemented them separately as it took me a while to grok the legacy implementation and this created two simpler flows.
Due to deadline I likely won't address this until post ship.